internal/attestation/measurements/measurements.go

```diff
@@ -94,58 +70,207 @@ func (m *M) FetchAndVerify(ctx context.Context, client *http.Client, measurement
+	if err := json.Unmarshal(b, &eM); err != nil {
+		// Unmarshalling failed, Measurement might be in legacy format,
+		// meaning a simple string instead of Measurement struct.
+		// TODO: remove with v2.4.0
+		if legacyErr := json.Unmarshal(b, &eM.Expected); legacyErr != nil {
+			return multierr.Append(
+				err,
+				fmt.Errorf("trying legacy format: %w", legacyErr),
+			)
+		}
+	}
+
+	if err := m.unmarshal(eM); err != nil {
+		return fmt.Errorf("unmarshalling json: %w", err)
 	}
+	return nil
+}
 
-	*m = make(M)
-	for key, value := range base64Map {
-		measurement, err := base64.StdEncoding.DecodeString(value)
+// MarshalJSON writes out a Measurement with Expected encoded as a hex string.
+func (m Measurement) MarshalJSON() ([]byte, error) {
+	return json.Marshal(encodedMeasurement{
+		Expected: hex.EncodeToString(m.Expected[:]),
+		WarnOnly: m.WarnOnly,
+	})
+}
+
+// UnmarshalYAML reads a Measurement either as yaml object,
+// or as a simple hex or base64 encoded string.
+func (m *Measurement) UnmarshalYAML(unmarshal func(any) error) error {
+	var eM encodedMeasurement
+	if err := unmarshal(&eM); err != nil {
+		// Unmarshalling failed, Measurement might be in legacy format,
+		// meaning a simple string instead of Measurement struct.
+		// TODO: remove with v2.4.0
+		if legacyErr := unmarshal(&eM.Expected); legacyErr != nil {
+			return multierr.Append(
+				err,
+				fmt.Errorf("trying legacy format: %w", legacyErr),
+			)
+		}
+	}
+
+	if err := m.unmarshal(eM); err != nil {
+		return fmt.Errorf("unmarshalling yaml: %w", err)
+	}
+	return nil
+}
+
+// MarshalYAML writes out a Measurement with Expected encoded as a hex string.
+func (m Measurement) MarshalYAML() (any, error) {
+	return encodedMeasurement{
+		Expected: hex.EncodeToString(m.Expected[:]),
+		WarnOnly: m.WarnOnly,
+	}, nil
+}
+
+// unmarshal parses a hex or base64 encoded Measurement.
+func (m *Measurement) unmarshal(eM encodedMeasurement) error {
+	expected, err := hex.DecodeString(eM.Expected)
+	if err != nil {
+		// expected value might be in base64 legacy format
+		// TODO: Remove with v2.4.0
+		hexErr := err
+		expected, err = base64.StdEncoding.DecodeString(eM.Expected)
 		if err != nil {
-			return err
+			return multierr.Append(
+				fmt.Errorf("invalid measurement: not a hex string %w", hexErr),
+				fmt.Errorf("not a base64 string: %w", err),
+			)
 		}
-		(*m)[key] = measurement
 	}
+
+	if len(expected) != 32 {
+		return fmt.Errorf("invalid measurement: invalid length: %d", len(expected))
+	}
+
+	m.Expected = *(*[32]byte)(expected)
+	m.WarnOnly = eM.WarnOnly
+
 	return nil
 }
 
+// WithAllBytes returns a measurement value where all 32 bytes are set to b.
+func WithAllBytes(b byte, warnOnly bool) Measurement {
+	return Measurement{
+		Expected: *(*[32]byte)(bytes.Repeat([]byte{b}, 32)),
+		WarnOnly: warnOnly,
+	}
+}
+
+// DefaultsFor provides the default measurements for given cloud provider.
+func DefaultsFor(provider cloudprovider.Provider) M {
```

Comment:
we need default (dummy) values for those that previously have been included in EnforcedMeasurements